Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -926,36 +942,32 @@ DirectoryBlobPartition::DirectoryBlobPartition(
   if (fst_callback)
     fst_callback(&nodes, &dol_node);
 
-  const u64 new_fst_address = SetDOL(std::move(dol_node), new_dol_address);
-  BuildFST(std::move(nodes), new_fst_address);
-}
+  const u64 new_fst_address = SetDOL(std::move(dol_node), new_dol_address, disc_header);
+  BuildFST(std::move(nodes), new_fst_address, disc_header);
 
-void DirectoryBlobPartition::SetDiscHeaderFromFile(const std::string& boot_bin_path)
-{
-  m_disc_header.resize(DISCHEADER_SIZE);
-  if (ReadFileToVector(boot_bin_path, &m_disc_header) < 0x20)
-    ERROR_LOG_FMT(DISCIO, "{} doesn't exist or is too small", boot_bin_path);
-
-  m_contents.AddReference(DISCHEADER_ADDRESS, m_disc_header);
+  disc_header = std::move(ExtractNodeToVector(&sys_nodes, &disc_header));
```

Comment:
You need to do this before you use `disc_header` in the calls to `SetDiscType`, `SetDOL` and `SetFST`. (The value that `disc_header` had earlier is invalid because you used `std::move` on it.)